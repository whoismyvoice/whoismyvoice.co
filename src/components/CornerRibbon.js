import React from 'react';

import style from '../styles/CornerRibbon.scss';

const CornerRibbon = () => {
  return <div className="corner-ribbon">
  <a
    href="http://health.whoismyvoice.com/"
    target="_blank"
  >
    Care about healthcare?
  </a>
</div>;
};

export default CornerRibbon;
