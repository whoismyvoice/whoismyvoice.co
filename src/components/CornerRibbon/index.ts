import Component from './CornerRibbon';

export default Component;
