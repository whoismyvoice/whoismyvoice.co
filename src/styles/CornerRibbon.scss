@import 'variables';

.corner-ribbon {
  background-color: $dark-grey;
  overflow: hidden;
  white-space: nowrap;
  position: absolute;
  right: -60px;
  top: 45px;
  z-index: 99;
  -webkit-transform: rotate(45deg);
  -moz-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  -o-transform: rotate(45deg);
  transform: rotate(45deg);
}

.corner-ribbon a {
  border: 1px solid $white;
  text-transform: uppercase;
  color: #fff;
  display: block;
  font: bold 11px 'Open Sans', Helvetica, Arial, sans-serif;
  margin: 2px 0;
  padding: 14px 50px;
  text-align: center;
  text-decoration: none;
}
