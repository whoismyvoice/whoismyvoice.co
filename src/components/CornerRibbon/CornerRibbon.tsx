import React from 'react';
import cx from 'classnames';

import '../../styles/CornerRibbon.scss';

interface Props {
  didSearch?: boolean;
}

class CornerRibbon extends React.Component<Props> {
  render() {
    const { didSearch } = this.props;

    const ribbonClasses = cx(['corner-ribbon'], {
      'corner-ribbon__hide': didSearch,
    });
    return (
      <div className={ribbonClasses}>
        <a
          href="http://health.whoismyvoice.com/"
          target="_blank"
          rel="noreferrer noopener"
        >
          Care about healthcare?
        </a>
      </div>
    );
  }
}

export default CornerRibbon;
