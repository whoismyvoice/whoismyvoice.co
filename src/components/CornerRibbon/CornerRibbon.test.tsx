import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';
import CornerRibbon from './CornerRibbon';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CornerRibbon />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly no props', () => {
  const tree = renderer.create(<CornerRibbon />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly didSearch=true', () => {
  const tree = renderer.create(<CornerRibbon didSearch={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly didSearch=false', () => {
  const tree = renderer.create(<CornerRibbon didSearch={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});
