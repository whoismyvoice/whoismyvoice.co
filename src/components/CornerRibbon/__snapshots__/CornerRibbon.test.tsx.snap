// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly didSearch=false 1`] = `
<div
  className="corner-ribbon"
>
  <a
    href="http://health.whoismyvoice.com/"
    rel="noreferrer noopener"
    target="_blank"
  >
    Care about healthcare?
  </a>
</div>
`;

exports[`renders correctly didSearch=true 1`] = `
<div
  className="corner-ribbon corner-ribbon__hide"
>
  <a
    href="http://health.whoismyvoice.com/"
    rel="noreferrer noopener"
    target="_blank"
  >
    Care about healthcare?
  </a>
</div>
`;

exports[`renders correctly no props 1`] = `
<div
  className="corner-ribbon"
>
  <a
    href="http://health.whoismyvoice.com/"
    rel="noreferrer noopener"
    target="_blank"
  >
    Care about healthcare?
  </a>
</div>
`;
